### PR TITLE
Feature/ab2d 3824 factor tracking

### PR DESCRIPTION
**Pull request by nflearl**
**JIRA Tickets:**

[AB2D-3824](https://jira.cms.gov/browse/AB2D-3824) - Description

***Related Tickets***
 
### What Does This PR Do?
It isolates job tracking to a single service so we can switch over to using a queue instead of in-memory.  This will allow lambdas to send messages to update the status.

It is a purely refactoring checking.  Everything should behave almost exactly like before.  One test had to be tweaked as we are batching stat update, not doing individual increments.

### What Should Reviewers Watch For?

Carefully verify that I have hooked all the correct places in which to update statistics.

### Security Implications

- [ ] This PR adds new software dependencies
- [ ] This PR modifies or invalidates our security controls
- [ ] This PR stores or transmits data that was not stored or transmitted before

- [ ] This PR requires additional review of its security implications for other reasons

### Submitter Checklist

- [x] This PR includes any required documentation changes, (`README`, changelog / release notes, website).
- [x] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments.
- [x] Code checked for PHI/PII exposure

### CODE REVIEW

File: worker/src/main/java/gov/cms/ab2d/worker/processor/JobProgressServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    private final JobRepository jobRepository;
+
+    private final Map<String, ProgressTracker>  progressTrackerMap = new HashMap<>(89);
+
+    public JobProgressServiceImpl(JobRepository jobRepository) {
+        this.jobRepository = jobRepository;
+    }
+
+    @Override
+    public void addMeasure(String jobId, JobMeasure measure, long value) {
```
[the remark this answers is not in the document]
Comment (nflearl) [reply worker/src/main/java/gov/cms/ab2d/worker/processor/JobProgressServiceImpl.java:30]: For the current in-memory version - no issue.  

For when I swap in using a real queue in a later story, absolutely a concern that will need to be addressed.

File: worker/src/main/java/gov/cms/ab2d/worker/processor/JobProgressServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+    @Value("${report.progress.db.frequency:100}")
+    private int reportProgressDbFrequency;
+
+    @Value("${report.progress.log.frequency:100}")
+    private int reportProgressLogFrequency;
+
+    private final JobRepository jobRepository;
+
+    private final Map<String, ProgressTracker>  progressTrackerMap = new HashMap<>(89);
```
[the remark this answers is not in the document]
Comment (nflearl) [reply worker/src/main/java/gov/cms/ab2d/worker/processor/JobProgressServiceImpl.java:23]: Magic number with great growth properties.  The default map size is a horrible 16.  As the Map may grow, 89 scales into very nice prime or nearly prime numbers allowing for optimal distributions.

[the remark this answers is not in the document]
Comment (nflearl) [reply worker/src/main/java/gov/cms/ab2d/worker/processor/JobProgressServiceImpl.java:23]: I'll add a cleanup mechanism in later checkins, most likely when the job state is updated to it's final form.  For now, ProgressTrackers are not that large and this isn't too wasteful of memory.

If you want me to add a TODO, I'll be happy to do so.

[the remark this answers is not in the document]
Comment (sb-wnyffenegger) [reply worker/src/main/java/gov/cms/ab2d/worker/processor/JobProgressServiceImpl.java:23]: Maybe a TODO would be useful. That or a note on a to be completed story

**Review by sb-wnyffenegger — APPROVED** [review]

**Review by sharshar-sb — APPROVED** [review]

**Pull request merged**